fn may_fail(fail) {
    if fail {
        return nil, "error";
    }
    return "success", nil;
}

fn main() {
    // The ? operator simplifies Go-style error handling.
    // By default, this will make the whole program crash if an error is encountered.
    let ok = may_fail(false)?;
    print(f"First result: {ok}");

    // Sometimes it is desirable to log or try to recover from the error.
    // An err block may be used for this purpose:
    let ok = may_fail(true) err {
        print(f"An error has occurred: {err}");
        return ...recovery();
    }?;
    return ok;
}
